src/main/java/Duke.java

```diff
@@ -1,10 +1,154 @@
+import java.util.Scanner;
+
 public class Duke {
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        showWelcomeMessage();
+        while (true) {
+            String userCommand = getUserInput();
+            if (userCommand.equals("bye")) {
+                showExitMessage();
+                break;
+            }
+            else if (userCommand.equals("list")) {
+                showList(tasks, taskCount);
+            }
+            else if (userCommand.startsWith("done")) {
+                setDone(tasks, userCommand);
+            }
+            else {
+                int lineSize = userCommand.length();
+                showAddTaskMessage();
+                if (userCommand.startsWith("todo")) {
+                    taskCount = addTodo(tasks, taskCount, userCommand, lineSize);
+                }
+                else if (userCommand.startsWith("deadline")) {
+                    taskCount = addDeadline(tasks, taskCount, userCommand, lineSize);
+                }
+                else if (userCommand.startsWith("event")) {
+                    taskCount = addEvent(tasks, taskCount, userCommand, lineSize);
+                }
+                else {
+                    taskCount = addTask(tasks, taskCount, userCommand);
+                }
+                showTaskCountMessage(taskCount);
+            }
+        }
+    }
+
+    private static String getUserInput() {
+        Scanner input = new Scanner(System.in);
+        return input.nextLine();
+    }
+
+    private static int addTask(Task[] tasks, int taskCount, String userCommand) {
+        for (Task task : tasks) {
+            tasks[taskCount] = new Task(userCommand);
+        }
+        echoTask(tasks, taskCount);
+        taskCount++;
+        return taskCount;
+    }
+
+    private static int addTodo(Task[] tasks, int taskCount, String userCommand, int lineSize) {
+        String todoTask = getTask(userCommand);
+        for (Task task : tasks) {
+
```

Comment:
removing space will help with readability (duplicate comment because I need to redo the review)